includes/class-wc-payments-woopay-direct-checkout.php

```diff
@@ -39,11 +42,23 @@ public function maybe_use_store_api_draft_order_id( $order_id ) {
 		$is_already_defined_order_id = ! empty( $order_id );
 		// Only apply this filter if the session doesn't already have an order_awaiting_payment.
 		$is_order_awaiting_payment = isset( WC()->session->order_awaiting_payment );
-		if ( ! $is_checkout || $is_already_defined_order_id || $is_order_awaiting_payment ) {
+		// Only apply this filter if draft order ID exists.
+		$is_draft_order_exists = ! empty( WC()->session->get( 'store_api_draft_order' ) );
+		if ( ! $is_checkout || $is_already_defined_order_id || $is_order_awaiting_payment || ! $is_draft_order_exists ) {
```

Comment:
> [!NOTE]
> This is a nitpick.

Can we call this variable either `$does_draft_order_exist` or `$has_draft_order`? The `is` at the start here makes it a bit weird to read 😅 

```suggestion
		$does_draft_order_exist = ! empty( WC()->session->get( 'store_api_draft_order' ) );
		if ( ! $is_checkout || $is_already_defined_order_id || $is_order_awaiting_payment || ! $does_draft_order_exist ) {
```

```suggestion
		$has_draft_order = ! empty( WC()->session->get( 'store_api_draft_order' ) );
		if ( ! $is_checkout || $is_already_defined_order_id || $is_order_awaiting_payment || ! $has_draft_order ) {
```